### ISSUE TITLE
[ISSUE] `Toga.Box()` on the `ROW` direction in RTL Android Phones is a mess!

### ISSUE THREAD

**Issue by vzool** [beeware/toga#2692]
### Describe the bug

`Toga.Box()` on the `ROW` direction in RTL Android Phones is a mess, while it is working just fine in LTR.
![WhatsApp Image 2024-07-07 at 6 17 34 AM](https://github.com/beeware/toga/assets/4952736/09ebe41e-a206-4caa-99ac-1aed36ec8d9d)

### Steps to reproduce

1. Clone the repo `git@github.com:vzool/zakat-tracker.git`
2. Run on Android Phone with RTL locale like (Arabic or Hebrew)
3. Browse the UI
4. See the error

### Expected behavior

The UI should behave the same on all RTL or LTR locale phones.

### Screenshots

On the left is the issue on RTL Android phone languages like Arabic, on the right is the same UI on LTR Android phone languages like English:
<img width="357" alt="‏لقطة الشاشة 1446-01-01 في 6 02 07 ص" src="https://github.com/beeware/toga/assets/4952736/d8a1d81a-4fe2-4540-95e7-29025e9a9090">
<img width="387" alt="‏لقطة الشاشة 1446-01-01 في 6 02 55 ص" src="https://github.com/beeware/toga/assets/4952736/0114e5e8-f43f-47e1-bfe5-11192bf99feb">
is 

### Environment

- Operating System: macOS Sonoma 14.5 (23F79)
- Python version: 3.12.4
- Software versions:
  - Briefcase: 0.3.19
  - Toga: 0.4.5

### Logs

```

```

**Comment by freakboy3742** [comment]
Thanks for the report. I don't doubt there are issues with RTL handling; sadly, none of the currently active core team members are fluent in an RTL language.

It would be especially helpful if you're able to provide examples of layouts that you know to be problematic, with code samples. There are parts of RTL handling that are implemented as cross-platform layout code, and some that leverage platform RTL handling. Having specific examples of problematic layouts provides specific issues that can be resolved, rather than a nebulous "it's a mess" descriptor which, while it may be true, doesn't provide a very good mechanism for us to determine when enough fixes have been applied that it is no longer "a mess".

**Comment by vzool** [comment]
Great, here is a snippet code illustrating the issue in a single file `app.py`:

```python
"""
My first application
"""

import toga
from toga.style import Pack
from toga.style.pack import COLUMN, ROW
from datetime import datetime

class HelloWorld(toga.App):
    def startup(self):
        """Construct and show the Toga application.

        Usually, you would add your application to a main content box.
        We then create a main window (with a name matching the app), and
        show the main window.
        """
        main_box = toga.Box()

        main_box.add(self.datetime_widget())

        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = main_box
        self.main_window.show()

    def datetime_widget(self):
        now = datetime.now()
        datetime_box = toga.Box(style=Pack(direction=COLUMN, padding=5))
        datetime_label = toga.Label('DateTime', style=Pack(padding=(0, 5), text_align="center", font_weight='bold'))

        year_label = toga.Label('Year', style=Pack(padding=(0, 5)))
        self.year_input = toga.NumberInput(min=1000, value=now.year, style=Pack(flex=1, width=66))
        month_label = toga.Label('Month', style=Pack(padding=(0, 5)))
        self.month_input = toga.NumberInput(min=1, max=12, value=now.month, style=Pack(flex=1, width=45))
        day_label = toga.Label('Day', style=Pack(padding=(0, 5)))
        self.day_input = toga.NumberInput(min=1, max=31, value=now.day, style=Pack(flex=1, width=45))

        date_box = toga.Box(style=Pack(direction=ROW, padding=5, flex=1))
        date_box.add(year_label)
        date_box.add(self.year_input)
        date_box.add(month_label)
        date_box.add(self.month_input)
        date_box.add(day_label)
        date_box.add(self.day_input)

        hour_label = toga.Label('Hour', style=Pack(padding=(0, 5)))
        self.hour_input = toga.NumberInput(min=0, max=23, value=now.hour, style=Pack(flex=1, width=45))
        minute_label = toga.Label('Minute', style=Pack(padding=(0, 5)))
        self.minute_input = toga.NumberInput(min=0, max=59, value=now.minute, style=Pack(flex=1, width=45))
        second_label = toga.Label('Second', style=Pack(padding=(0, 5)))
        self.second_input = toga.NumberInput(min=0, max=59, value=now.second, style=Pack(flex=1, width=45))

        time_box = toga.Box(style=Pack(direction=ROW, padding=5, flex=1))
        time_box.add(hour_label)
        time_box.add(self.hour_input)
        time_box.add(minute_label)
        time_box.add(self.minute_input)
        time_box.add(second_label)
        time_box.add(self.second_input)

        datetime_box.add(datetime_label)
        datetime_box.add(toga.Divider())
        datetime_box.add(date_box)
        datetime_box.add(time_box)
        return datetime_box

def main():
    return HelloWorld()

```
# Normal behavior on Android LTR locale device:
![Screenshot_1720436954](https://github.com/beeware/toga/assets/4952736/f0a190b4-dab8-4bee-8c2e-af6d3daf317c)
# Issue caught again on Android RTL locale device:
![Screenshot_1720437060](https://github.com/beeware/toga/assets/4952736/f9e3192a-d1b9-448b-9f38-104adb6942ca)

**Thanks**

**Comment by mhsmith** [comment]
I see the same on API level 34. It can be worked around by editing build/.../android/gradle/app/src/main/AndroidManifest.xml to change `android:supportsRtl="false"` to `"true"`.

We should probably make this change in the template until we support RTL layouts properly.

**Comment by mhsmith** [comment]
Actually we shouldn't make this change in the Briefcase template, because it would be difficult to remove in the future given that there's not a direct relationship between Toga and Briefcase versions. Instead, we should try and fix Toga so it's at least usable in RTL locales until we support them fully.

**Comment by vzool** [comment]
Thanks for the suggestion.
But, I searched the BeeWare repositories and I saw that `android:supportsRtl="true"` at [AndroidManifest.xml](https://github.com/beeware/briefcase-android-gradle-template/blob/1c5557520f2eebf1054f8ec148a581a9a8cc3631/%7B%7B%20cookiecutter.format%20%7D%7D/app/src/main/AndroidManifest.xml) is already set by the [briefcase-android-gradle-template](https://github.com/beeware/briefcase-android-gradle-template) on this [line](https://github.com/beeware/briefcase-android-gradle-template/blob/1c5557520f2eebf1054f8ec148a581a9a8cc3631/%7B%7B%20cookiecutter.format%20%7D%7D/app/src/main/AndroidManifest.xml#L15), as I understand the UI is RTL but Toga content can't adapt.
I think Android RTL/LTR locales will force its direction, no matter what was set for `android:supportsRtl="true"`.
**Android**: If the user wants RTL/LTR, your wish is my command. ^_^

**Comment by mhsmith** [comment]
> I think Android RTL/LTR locales will force its direction, no matter what was set for `android:supportsRtl="true"`

Sorry, I got that the wrong way round. Try setting it to "false".

I expect this will still have some issues in the presence of Arabic text, so please try it and let us know what you think.

**Comment by vzool** [comment]
Thank you very much, it is way better now.
BTW, will you accept PR if I make this change on [briefcase-android-gradle-template](https://github.com/beeware/briefcase-android-gradle-template)?

**Comment by mhsmith** [comment]
No, I changed my mind about that – see [this comment](https://github.com/beeware/toga/issues/2692#issuecomment-2215448727).
